x-pack/legacy/plugins/siem/public/pages/detection_engine/rules/components/pick_timeline/index.tsx

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { EuiFormRow } from '@elastic/eui';
+import React, { useCallback, useEffect, useState } from 'react';
+// import styled from 'styled-components';
+
+import { SearchTimelineSuperSelect } from '../../../../../components/timeline/search_super_select';
+import { FieldHook, getFieldValidityAndErrorMessage } from '../shared_imports';
+
+export interface FieldValueTimeline {
+  id: string | null;
+  title: string | null;
+}
+
+interface QueryBarDefineRuleProps {
+  dataTestSubj: string;
+  field: FieldHook;
```

Comment:
nit (since I saw you editing `es_ui_shared above` ;): Consider adding generics to the `FieldHook` type so you can declare it as:

``` ts
field: FieldHook<FieldValueTimeline>;
``` 

which then removes the need for the `as` cast on `38`.